src/filebrowser/PasMetadata.vue

```diff
@@ -0,0 +1,240 @@
+
+				// remove csv info if not sequential
+				delete payload.file_characteristics.csv_delimiter;
+				delete payload.file_characteristics.csv_has_header;
+				delete payload.file_characteristics.csv_record_separator;
+				delete payload.file_characteristics.csv_quoting_char;
+			}
+
+			return payload;
+		},
+		// needs to support adding values and updating values. Should be able to delete csv values
+		async save() {
+			try {
+				this.saveResult = null; // reset save failed/success alert
+				const { data: { success, failed } } = await axios.patch('files/', [this.payload], {
+					baseURL: process.env.VUE_APP_METAX_FILEAPI_URL,
+				});
+				if (failed.length > 0) {
+					this.saveResult = 'failed';
+				} else {
+					this.saveResult = 'success';
+					this.dataFromServer = success[0].object; // set ui values based on save result
+				}
+			} catch (error) {
+				this.saveResult = 'failed';
+				throw error;
+			}
+
+		}
+	},
+	data() {
+		return {
+			dataFromServer: null,
+			saveResult: null,
+			sequential: false,
+			separatorOptions: [
+				{ value: 'LF', text: 'LF (default)' },
+				{ value: 'CRLF', text: 'CRLF' },
+				{ value: 'CR', text: 'CR' },
+				/* CR LF (Windows), LF (Unix) and CR (Macintosh) line break types.
+					The Carriage Return (CR) character (0x0D, \r) moves the cursor to the beginning of the line without advancing to the next line.
+					This character is used as a new line character in Commodore and Early Macintosh operating systems (OS-9 and earlier).
+
+					The Line Feed (LF) character (0x0A, \n) moves the cursor down to the next line without returning to the beginning of the line.
+					This character is used as a new line character in UNIX based systems (Linux, Mac OSX, etc)
+
+					The End of Line (EOL) sequence (0x0D 0x0A, \r\n) is actually two ASCII characters, a combination of the CR and LF characters.
+					It moves the cursor both down to the next line and to the beginning of that line.
+					This character is used as a new line character in most other non-Unix operating systems including Microsoft Windows, Symbian OS and others.
+
+					Source: https://stackoverflow.com/questions/1552749/difference-between-cr-lf-lf-and-cr-line-break-types
+				*/
+			],
+			delimiterOptions: [
+				{ value: '\t', text: 'Tab' },
+				{ value: ' ', text: 'Space' },
+				{ value: ';', text: 'Semicolon' },
+				{ value: ',', text: 'Comma (default)' },
+				{ value: ':', text: 'Colon' },
+				{ value: '.', text: 'Dot' },
+				{ value: '|', text: 'Pipe' }
+			],
+			encodingOptions: [
+				{ value: 'UTF-8', text: 'UTF-8 (default)' },
+				{ value: 'UTF-16', text: 'UTF-16' },
+				{ value: 'UTF-32', text: 'UTF-32' },
+				{ value: 'ISO-8859-15', text: 'ISO-8859-15' },
+			],
+		}
+	},
+	created() {
+		// set values to data for editability. Use Object.assign to avoid sharing object references with parent.
+		this.dataFromServer = Object.assign({}, this.file);
```

Comment:
this does not work as expected. See Deep Copy notes in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

Reply:
In the next line the file_charasteristics is copied to the same object. That is the only nested object that we are going to alter in this component. So we don't need a complete deep copy.

Reply:
I would prefer to have `JSON.parse(JSON.stringify(obj1))` pattern there as then it would be future proof as well, but the minimum is to comment the next line there, why it is done, then it is clear.

Reply:
The future proof is good argument. I will update the code.